src/basic/time-util.c

```diff
@@ -55,6 +56,20 @@ usec_t now(clockid_t clock_id) {
         return timespec_load(&ts);
 }
 
+usec_t now_reproducible(clockid_t clock_id) {
+        struct timespec ts;
+        int r;
+        unsigned long long epoch;
+
+        r = getenv_llu_secure("SOURCE_DATE_EPOCH", 10, &epoch);
+        if (r == 0)
+                return (usec_t) epoch * USEC_PER_SEC;
+
+        assert_se(clock_gettime(map_clock_id(clock_id), &ts) == 0);
+
+        return timespec_load(&ts);
```

Comment:
Why not simply write `return now(clock_id);`??

Reply:
Thanks a lot for your very valuable input! I think I resolved all your points accordingly.